File: app/src/main/java/com/devx/raju/ui/custom/recyclerview/RecyclerLayoutClickListener.java
package com.devx.raju.ui.custom.recyclerview;

import android.view.View;

import com.devx.raju.data.local.entity.GithubEntity;

public interface RecyclerLayoutClickListener {

    void redirectToDetailScreen(View imageView,
                                View titleView,
                                View revealView,
                                View languageView,
                                GithubEntity githubEntity);

    void sharePost(GithubEntity githubEntity);
}

File: app/src/main/java/com/devx/raju/ui/custom/recyclerview/RecyclerLayoutClickListener.kt
package com.devx.raju.ui.custom.recyclerview

import android.view.View
import com.devx.raju.data.local.entity.GithubEntity

interface RecyclerLayoutClickListener {
    fun redirectToDetailScreen(imageView: View,
                               titleView: View,
                               revealView: View,
                               languageView: View,
                               githubEntity: GithubEntity)

    fun sharePost(githubEntity: GithubEntity)
}